File: python/misc/binary_watch.py
class BinaryWatch:
    def __init__(self):
        self._cache = {}
        for h in range(12):
            for m in range(60):
                b = self.__bits__(h, m)
                t = self.__i_to_time__(h, m)
                if b in self._cache:
                    self._cache[b] += [t]
                else:
                    self._cache[b] = [t]

    def times(self, n):
        if n in self._cache:
            return self._cache[n]
        else:
            return []

    def __bits__(self, h, m):
        b = 0

        while h > 0:
            b += 1
            h -= h & -h

        while m > 0:
            b += 1
            m -= m & -m

        return b

    def __i_to_time__(self, h, m):
        return "{:d}:{:02d}".format(h, m)

File: javascript/js/misc/binary_watch.js
class BinaryWatch {
    constructor() {
        this.cache = new Map()
        for (let h = 0; h < 12; h++) {
            for (let m = 0; m < 60; m++) {
                const b = bits(h, m)
                const s = formatHM(h, m)
                if (this.cache.has(b)) {
                    this.cache.get(b).push(s)
                } else {
                    this.cache.set(b, [s])
                }
            }
        }
    }

    bits(h, n) {
        var b = 0;

        while (h > 0) {
            h -= h & -h;
            b += 1;
        }

        while (m > 0) {
            m -= m & -m;
            b += 1;
        }

        return b
    }

    formatHM(h, m) {
        const hs = h.toString();
        const ms = m.toString();
        const fms = ms.length() == 1 ? "0" : "" + ms
        return `${hs}:${fms}`;
    }

    watchN(n) {
        if (n < 0 || n > 8) {
            return [];
        } else {
            return this.cache.get(n);
        }
    }
}